tests/test_cli.py

```diff
@@ -74,8 +75,11 @@ def test_encrypt_decrypt_submission(
         temp_config_file_path,
     ]
 
-    runner = CliRunner()
-    result = runner.invoke(grz_cli.cli.cli, testargs, catch_exceptions=False)
+    # for some reason, passing `env={…}` to CliRunner is not enough, so setting os.environ here as well.
+    os.environ["GRZ_MODE"] = "true"
+    runner = CliRunner(env={"GRZ_MODE": "true"})
+    cli = grz_cli.cli.build_cli()
```

Comment:
The reason is that you're running `cli = grz_cli.cli.build_cli()` outside the CliRunner in line 81 I think.
What about using `@mock.patch.dict(os.environ, {"GRZ_MODE": "true"})`?